src/cljs/rems/spa.cljs

```diff
@@ -411,13 +399,32 @@
 ;; must be called after routes have been defined
 
 (defn hook-browser-navigation! []
-  (doto (History.)
-    (events/listen
-     HistoryEventType/NAVIGATE
-     (fn [event]
-       (js/window.rems.hooks.navigate (.-token event))
-       (secretary/dispatch! (.-token event))))
-    (.setEnabled true)))
+  (events/listen accountant/history
+                 HistoryEventType/NAVIGATE
+                 (fn [event]
+                   (js/window.rems.hooks.navigate (.-token event))))
+  (accountant/configure-navigation!
+   {:nav-handler (let [previous-path (atom nil)]
+                   (fn [path]
+                     ;; XXX: workaround for Secretary/Accountant considering URLs with different hash to be different pages
+                     (let [url (js/URL. path js/location)
+                           path-without-hash (str (.-pathname url) (.-search url))]
+                       (when-not (= @previous-path path-without-hash)
+                         (reset! previous-path path-without-hash)
+                         (secretary/dispatch! path-without-hash)))))
+    :path-exists? (fn [path]
+                    (let [route (secretary/locate-route path)
+                          not-found-page? (= "*" (secretary/route-value (:route route)))]
```

Comment:
do we need the not-found-page route if we actively avoid it here?

Reply:
This avoidance is to tell Accountant whether to use SPA navigation for a particular link. Otherwise it'll do a traditional full page reload. The not-found page is still needed when the user arrives to the site using a broken link.